server/src/main/java/org/opensearch/extensions/rest/RestSendToExtensionAction.java

```diff
@@ -259,7 +251,12 @@ public String executor() {
                     filteredHeaders,
                     contentType,
                     content,
-                    requestIssuerIdentity,
+                    identityService.getTokenManager()
```

Comment:
Lets put this on its own line.

It looks like the only value that is being included in the claim for this token is the audience / application id,  Could we change the function signature to strongly use types to make sure its usage is clear `issueOnBehalfOfToken(..., final Principal applicationPrincipal)`?

Reply:
I was thinking we would implicitly override this handling by making it so that if the audience matched the issuer we got the service account token here. Perhaps that is not required or not the way forward but that was the idea I had been messing with over in the security plugin demo implementation I wrote up. It is also the fastest way to address the need since it requires only a single change. 

If you think that is not the way to go, we can address the service account token handling later and ask Ryan to remove the abstraction here if he gets to working on this further. Otherwise I will make the swaps next week.

Reply:
Making this just take a tokenManager and then will have the operations in the body itself.